lib/import-export-visitor.js

```diff
@@ -402,48 +401,40 @@ function getBlockBodyInfo(visitor, path) {
   const node = path.getNode();
   const parent = path.getParentNode() || node;
 
-  let needToAddUseStrictDirective = false;
-  let insertCharIndex = node.start;
+  let body = parent.body;
   let bodyName = "body";
-  let body;
+  let insertCharIndex = node.start;
+  let needToAddUseStrictDirective = false;
 
   switch (parent.type) {
   case "Program":
-    body = parent.body;
     insertCharIndex = parent.start;
 
     // If parent is a Program, we may need to add a "use strict"
     // directive to enable const/let in Node 4.
     needToAddUseStrictDirective = visitor.enforceStrictMode;
-
     break;
 
   case "BlockStatement":
-    body = parent.body;
-
-    if (hasOwn.call(parent, "start")) {
-      insertCharIndex = parent.start + 1;
-    } else {
-      insertCharIndex = body[0].start;
-    }
+    insertCharIndex = parent.start + 1;
     break;
 
   case "SwitchCase":
-    bodyName = "consequent";
     body = parent.consequent;
+    bodyName = "consequent";
     insertCharIndex = body[0].start;
     break;
 
   default:
-    bodyName = path.getName();
-
     const block = {
       type: "BlockStatement",
-      body: []
+      body: [],
+      start: node.start,
+      end: node.end + 2
```

Comment:
Since we're using the start/end information to modify the original source, they should point to a correct position in the source code, not the final generated code. If you think about it that way, the start/end values of the final AST aren't wrong after modifying the AST; they just still refer to the original source.